### PR TITLE
GFORMS-1976: Add functionality to the toolbox link in test environmen…

### PR DESCRIPTION
**Pull request by lxol**
…ts to export and import form data from/to user session

### CODE REVIEW

File: app/uk/gov/hmrc/gform/ApplicationLoader.scala [hunk 1]
```diff
@@ -181,52 +182,56 @@ class ApplicationModule(context: Context)
...
-            Indexes.ascending("submitDetails.createdAt"),
-            IndexOptions()
-              .background(false)
-              .name("submittedIndex")
-              .expireAfter(submittedExpiry, TimeUnit.MILLISECONDS)
-          )
-        )
-        MongoUtils.ensureIndexes(this.collection, indexes, true)
-      }
-    },
```
Comment (josef-vlach) [inline app/uk/gov/hmrc/gform/ApplicationLoader.scala:222]: We should not be reusing this for snapshot data.

File: app/uk/gov/hmrc/gform/ApplicationLoader.scala [hunk 2]
```diff
@@ -272,13 +277,17 @@ class ApplicationModule(context: Context)
     )
   private val obligationModule = new ObligationModule(wSHttpModule, configModule)
   private val employmentsModule = new EmploymentsModule(wSHttpModule, configModule)
+
+  val savedFormsMongoCache = createMongoCacheRepository("savedForms")
```
Comment (josef-vlach) [inline app/uk/gov/hmrc/gform/ApplicationLoader.scala:281]: `savedForms` is overloaded name, let's call this `snapshots`

File: app/uk/gov/hmrc/gform/testonly/SavedForm.scala [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+import play.api.libs.json.OFormat
+import play.api.libs.json.Reads._
+import play.api.libs.json._
+
+case class SavedForm(
+  templateId: String,
+  formId: String,
+  userId: String,
+  savedAt: Instant
+)
```
Comment (josef-vlach) [inline app/uk/gov/hmrc/gform/testonly/SavedForm.scala:31]: Let call this `Snaphost`. 

Why this stores `formId` and `userId`?

It should store the raw data in something like `Map[String, String]`. Perhaps we can reuse `FormData`.

**Review by josef-vlach — CHANGES_REQUESTED** [review]

File: app/uk/gov/hmrc/gform/ApplicationLoader.scala (hunk 1 above)
Comment (lxol) [reply app/uk/gov/hmrc/gform/ApplicationLoader.scala:222]: We don't reuse `form` repository for snapshot data. It is just refactoring. Production code is not being changed. We just use the production instance of the `form` repository in the TestOnlyFormService to copy/restore to/from snapshot repository.

File: app/uk/gov/hmrc/gform/testonly/TestOnlyFormService.scala [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+  jsonCrypto: Encrypter with Decrypter
+)(implicit ec: ExecutionContext) {
+
+  def saveForm(saveRequest: SaveRequest): Future[SaveReply] =
+    formRepository.findById(saveRequest.formId).flatMap {
+      case Some(cacheItem) =>
+        val newId = java.util.UUID.randomUUID().toString
+        cacheItem.data.fields.toList
+          .traverse { case (k, v) =>
+            val value = if (k == "form") { v.as[JsObject] - "_id" - "userId" }
```
Comment (DursunGundogan) [inline app/uk/gov/hmrc/gform/testonly/TestOnlyFormService.scala:41]: Can we use `===` instead of `==`?

**Review by josef-vlach — APPROVED** [review]

**Pull request merged**